src/main/java/org/opensearch/knn/training/TrainingJob.java

```diff
@@ -103,38 +108,64 @@ public Model getModel() {
 
     @Override
     public void run() {
-        NativeMemoryAllocation nativeMemoryAllocation = null;
+        NativeMemoryAllocation trainingDataAllocation = null;
+        NativeMemoryAllocation modelAnonymousAllocation = null;
         ModelMetadata modelMetadata = model.getModelMetadata();
 
+        try {
+            // Reserve space in the cache for the model
+            modelAnonymousAllocation = nativeMemoryCacheManager.get(modelAnonymousEntryContext, false);
+
+            // Lock until training completes
+            modelAnonymousAllocation.readLock();
```

Comment:
Lets reverse the order. First we can check for training allocation and then check for model allocation. Its possible model allocation fits but training allocation does not and in this case model allocation might be redundant.